src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx

```diff
@@ -112,13 +126,44 @@ function BaseOnboardingAccounting({shouldUseNativeStyles}: BaseOnboardingAccount
                 text={translate('common.confirm')}
                 // eslint-disable-next-line rulesdir/prefer-early-return
                 onPress={() => {
-                    if (!userReportedIntegration) {
+                    if (userReportedIntegration === undefined) {
                         setError(translate('onboarding.purpose.errorSelection'));
                         // eslint-disable-next-line no-useless-return
                         return;
                     }
 
-                    // TODO: call CompleOnboarding API after the back-end PR is complete
+                    if (!onboardingPurposeSelected) {
+                        return;
+                    }
+
+                    if (onboardingPolicyID) {
+                        Policy.enablePolicyConnections(onboardingPolicyID, true);
+                    }
+
+                    Report.completeOnboarding(
+                        onboardingPurposeSelected,
+                        CONST.ONBOARDING_MESSAGES[onboardingPurposeSelected],
+                        {
+                            firstName: '',
+                            lastName: '',
+                        },
```

Comment:
This should be optional.

(To recheck: the BE seems to complain about the empty names)

Reply:
Should we also update the default value of this data in `compleOnboarding` function with the default value as empty string?

Reply:
Make it optional in completeOnboarding

Reply:
AFAICT it is optional. Lmk what you are seeing. Yes I would update this to be optional.

I am also pretty confused about why it's being passed as an object if we can fix that while we are here. Maybe someone thought it better to parameterize the arguments? But the next person disagreed 😄  

Looks inconsistent to me. Let's make it:

```js
function completeOnboarding(
    engagementChoice: OnboardingPurposeType,
    data: ValueOf<typeof CONST.ONBOARDING_MESSAGES>,
    firstName?: string;
    lastName?: string;
    adminsChatReportID?: string,
    onboardingPolicyID?: string,
    paymentSelected?: string,
    ...
) {
```

Reply:
@marcaaron 

```js
{
    "jsonCode": 402,
    "message": "402 Missing firstName",
    "onyxData": [],
    "requestID": "8c542209db944e20-ORN"
}
```

Reply:
Sigh... lol 😄

Reply:
@marcaaron Should we add default value for `firstName` and `lastName` in `completeOnboarding` because in the type `CompleteGuidedSetupParams`, `firstName` and `lastName` is string.

Reply:
Yes let's do it. We will need it in some cases, but not for the "manage team expense" choice so it is appropriate to make optional.

Reply:
I updated.